dotnet/src/SemanticKernel.Core/Functions/SKPlugin.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.SemanticKernel.Diagnostics;
+
+#pragma warning disable IDE0130
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>Provides an <see cref="ISKPlugin"/> implementation around a collection of functions.</summary>
+[DebuggerDisplay("Name = {Name}, Functions = {FunctionCount}")]
+[DebuggerTypeProxy(typeof(SKPlugin.TypeProxy))]
+public class SKPlugin : ISKPlugin
+{
+    /// <summary>The collection of functions associated with this plugin.</summary>
+    private readonly Dictionary<string, ISKFunction> _functions;
+
+    /// <summary>Initializes the new plugin from the provided name.</summary>
+    /// <param name="name">The name for the plugin.</param>
+    public SKPlugin(string name) : this(name, description: null, functions: null)
+    {
+    }
+
+    /// <summary>Initializes the new plugin from the provided name and function collection.</summary>
+    /// <param name="name">The name for the plugin.</param>
+    /// <param name="functions">The initial functions to be available as part of the plugin.</param>
+    public SKPlugin(string name, IEnumerable<ISKFunction>? functions) : this(name, description: null, functions)
+    {
+    }
+
+    /// <summary>Initializes the new plugin from the provided name, description, and function collection.</summary>
+    /// <param name="name">The name for the plugin.</param>
+    /// <param name="description">A description of the plugin.</param>
+    /// <param name="functions">The initial functions to be available as part of the plugin.</param>
+    public SKPlugin(string name, string? description, IEnumerable<ISKFunction>? functions = null)
+    {
+        Verify.ValidPluginName(name);
+
+        this.Name = name;
+        this.Description = !string.IsNullOrWhiteSpace(description) ? description! : "";
+
+        this._functions = new Dictionary<string, ISKFunction>(StringComparer.OrdinalIgnoreCase);
+        if (functions is not null)
+        {
+            foreach (ISKFunction f in functions)
+            {
+                Verify.NotNull(f, nameof(functions));
+                this._functions.Add(f.Name, f);
+            }
+        }
+    }
+
+    /// <inheritdoc/>
+    public string Name { get; }
+
+    /// <inheritdoc/>
+    public string Description { get; }
+
+    /// <summary>Gets the number of functions in this plugin.</summary>
+    public int FunctionCount { get; }
```

Comment:
It seems the property was forgotten :) - `public int FunctionCount => this._functions.Count; ` ?

Reply:
Oops! Thanks.